dj/models/node.py

```diff
@@ -152,6 +152,44 @@ class NodeMissingParents(SQLModel, table=True):  # type: ignore
     )
 
 
+class AvailabilityStateBase(SQLModel):
+    """
+    An availability state base
+    """
+
+    catalog: Optional[str] = None
+    schema_: Optional[str] = Field(default=None)
+    table: str
+    valid_through_ts: int
+    max_partition: List[str] = Field(sa_column=SqlaColumn(JSON))
+    min_partition: List[str] = Field(sa_column=SqlaColumn(JSON))
+
+
+class AvailabilityState(AvailabilityStateBase, table=True):  # type: ignore
```

Comment:
Is this `AvailabilityState` something that ought to have logic in some method that can just say `True` or `False` with maybe some criterion to say "you can use me" or not?

Reply:
Eventually yes but that's part of the downstream features that can be built on top of this information. For now this PR just facilitates external materializations announcing information on incremental materializations, i.e of only a single or a couple partitions. The external system then doesn't have to store the materialization state of each node. What it can instead do is just hit the DJ server, see that the max_partition is 20220120, and then materialize only the partitions for 20220120 to "today". Then push the new availability information to DJ so that it can merge it into the current availability state.

Reply:
Agree with @samredai and I think we will only need to add a route/method to scratch the current state by simply deleting it, when someone say, makes a change to a node... And the moment I wrote the previous sentence down I started to think... but wait, what if the user creates a new version of a node and wants to put it on a new table. Perhaps in that case we don't need to invalidate the old node's data availability state. Good topic for our next brainstorming session.